tests/test_oauth2client.py

```diff
@@ -0,0 +1,157 @@
+    old_class = oauth2client.service_account._JWTAccessCredentials
+    old_credentials = old_class.from_json_keyfile_name(
+        SERVICE_ACCOUNT_JSON_FILE)
+
+    new_credentials = _oauth2client._convert_service_account_credentials(
+        old_credentials)
+
+    assert (new_credentials._service_account_email ==
+            old_credentials.service_account_email)
+    assert new_credentials._signer.key_id == old_credentials._private_key_id
+    assert new_credentials._token_uri == old_credentials.token_uri
+
+
+def test__convert_gce_app_assertion_credentials():
+    old_credentials = oauth2client.contrib.gce.AppAssertionCredentials(
+        email='some_email')
+
+    new_credentials = _oauth2client._convert_gce_app_assertion_credentials(
+        old_credentials)
+
+    assert (new_credentials._service_account_email ==
+            old_credentials.service_account_email)
+
+
+@pytest.fixture
+def mock_oauth2client_gae_imports(mock_non_existent_module):
+    mock_non_existent_module('google.appengine.api.app_identity')
+    mock_non_existent_module('google.appengine.ext.ndb')
+    mock_non_existent_module('google.appengine.ext.webapp.util')
+    mock_non_existent_module('webapp2')
+
+
+@mock.patch('google.auth.app_engine.app_identity')
+def test__convert_appengine_app_assertion_credentials(
+        app_identity, mock_oauth2client_gae_imports):
+
+    import oauth2client.contrib.appengine
+
+    service_account_id = 'service_account_id'
+    old_credentials = oauth2client.contrib.appengine.AppAssertionCredentials(
+        scope='one two', service_account_id=service_account_id)
+
+    new_credentials = (
+        _oauth2client._convert_appengine_app_assertion_credentials(
+            old_credentials))
+
+    assert new_credentials.scopes == ['one', 'two']
+    assert (new_credentials._service_account_id ==
+            old_credentials.service_account_id)
+
+
+class MockCredentials(object):
+    pass
+
+
+def test_convert_success():
+    convert_function = mock.Mock()
+    conversion_map_patch = mock.patch.object(
+        _oauth2client, '_CLASS_CONVERSION_MAP',
+        {MockCredentials: convert_function})
+    credentials = MockCredentials()
+
+    with conversion_map_patch:
+        result = _oauth2client.convert(credentials)
+
+    convert_function.assert_called_once_with(credentials)
+    assert result == convert_function.return_value
+
+
+def test_convert_not_found():
+    with pytest.raises(ValueError) as excinfo:
+        _oauth2client.convert('a string is not a real credentials class')
+
+    assert excinfo.match('Unable to convert')
+
+
+def test_convert_no_oauth2client(monkeypatch):
+    monkeypatch.setattr(_oauth2client, '_HAS_OAUTH2CLIENT', False)
+
+    with pytest.raises(EnvironmentError) as excinfo:
+        _oauth2client.convert('any value will do here')
+
+    assert excinfo.match('oauth2client is not installed')
+
+
+def test_import_has_app_engine(mock_oauth2client_gae_imports):
+    reload_module(_oauth2client)
```

Comment:
Don't you need some kind of tear-down to put the module back the way it was before you reloaded it with `sys.modules` mocked?